src/trans.rs

```diff
@@ -482,47 +650,50 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
                             let fn_sig;
                             let mut call_kind = BinaryenCallKind::Direct;
 
-                            if fn_name == "wasm::::print_i32" || fn_name == "wasm::::_print_i32" {
-                                // extern wasm functions
-                                fn_sig = sig.clone();
-                                call_kind = BinaryenCallKind::Import;
-                                self.import_wasm_extern(fn_did, sig);
-                            } else {
-                                let nid = self.tcx.map.as_local_node_id(fn_did).expect("");
-                                let (nid, substs, sig) = if self.mir_map.map.contains_key(&nid) {
-                                    (nid, *substs, sig)
-                                } else {
-                                    // only trait methods can have a Self parameter
-                                    if substs.self_ty().is_none() {
-                                        panic!("unimplemented fn trans: {:?}", fn_did);
-                                    }
-
-                                    let (resolved_def_id, resolved_substs) = traits::resolve_trait_method(self.tcx, fn_did, substs);
-                                    let nid = self.tcx.map.as_local_node_id(resolved_def_id).expect("");
-                                    let ty = self.tcx.lookup_item_type(resolved_def_id).ty;
-                                    let sig = ty.fn_sig().skip_binder();
-
-                                    fn_did = resolved_def_id;
-                                    (nid, resolved_substs, sig)
-                                };
-
-                                let mir = &self.mir_map.map[&nid];
-
-                                fn_sig = monomorphize::apply_param_substs(self.tcx, substs, sig);
-                                {
-                                    let mut ctxt = BinaryenFnCtxt {
-                                        tcx: self.tcx,
-                                        mir_map: self.mir_map,
-                                        mir: mir,
-                                        did: fn_did,
-                                        sig: &fn_sig,
-                                        module: self.module,
-                                        fun_types: &mut self.fun_types,
-                                        fun_names: &mut self.fun_names,
-                                        c_strings: &mut self.c_strings,
+                            match fn_name.as_ref() {
+                                "wasm::::print_i32" | "wasm::::_print_i32" => {
```

Comment:
yes, as a temporary measure for testing, most spec impls provide the spectest module with those print fns AFAICT